src/image-copy/azext_imagecopy/custom.py

```diff
@@ -131,11 +131,16 @@ def imagecopy(source_resource_group_name, source_object_name, target_location,
 
         tasks = []
         for location in target_location:
+            # Appending location to target name if multiple locations
+            final_target_name = target_name
```

Comment:
target_name is optional. I think this new code block should only work when the value isn't None. 
What do you think?

Reply:
Do you mean to add another 'if' block to that?

Reply:
yes

Reply:
> 
> 
> yes

But next tasks.append function calling with final_target_name as argument and it could be undefined, so that why I unconditionally assign it.

Reply:
If target_name wasn't set by the user, meaning it's None, something will break - either on this method or in the create_target one.

You have 2 option:
1. start by setting final_target_name to target_name and then build the if around final_target_name
2. remove final_target_name and just overwrite target_name when appropriate

If it were me, I'd go with the second option. But both options should work IMO.

Reply:
Modifying of original target_name will have effect on next iteration how you suggest to resolve that?

Reply:
@tamirkamara so I do not understand what it breaking now,
If target_name is None, then final_target_name will also None and passed to array.
If target_name is set and count of target location is one, then it will pass original name to array as well.
If target_name and count of target location more then one, if will append suffix.
final_target_name - is temporary variable just to not modify target_name over iterations.